Exercise_4.java

```diff
@@ -32,6 +34,27 @@ static void inorder(Node temp)
     /*function to insert element in binary tree */
     static void insert(Node temp, int key) 
     { 
+    	Queue<Node> q = new LinkedList<Node>();

```

Comment:
This will never enter the while loop, you never added temp to the queue.
The queue will be empty and the program will terminate.